### ISSUE TITLE
Attribute "Recommended for kids"

### ISSUE THREAD

**Issue by andrixnet** [opencaching/opencaching-pl#806]
Attribute "Recommended for kids" is listed as having id=41.
According to OKAPI, OCDE has the same attribute with id=59 (unused by OCPL-current)
The result is two A code mappings in OKAPI for the same attribute. 

With the entry of OCUK to OCPL-current, OCUK brings many attributes that differ from OCPL-current, some similar to OCDE, some completely new.
OCUK already has attribute "Not available at high tide" id=41. Knowing the overlap (thanks to OKAPI), this can be easily resolved quickly and elegantly: 

According to http://opencaching.pl//okapi/devel/attrlist and http://www.opencaching.nl/okapi/devel/attrlist there is no attribute id=59 in use. 

OCPL and OCNL:
Run the following queries on your servers (and in devel machines) :  
```
update cache_attrib set id=59 where id=41;  
update caches_attributes set attrib_id=59 where attrib_id=41;
```

**Comment by andrixnet** [comment]
OKAPI fix requested by https://github.com/opencaching/okapi/issues/445

**Comment by andrixnet** [comment]
There are more attribute id duplications and overlaps, but for the time being, these area the easies to solve. (and among the ones with priority for integrating OCUK properly)

**Comment by deg-pl** [comment]
I think is MUCH easier to change one site - OC DE, not 4 different sites (NL, RO, PL, UK)

**Comment by andrixnet** [comment]
I can handle RO, help Harrie with NL, you can handle UK and (probably) PL. 
Anyway, AFAIK things seem to happen much faster at OCPL development then OCDE.

**Comment by deg-pl** [comment]
I'm afraid malfunctioning OKAPI during changes. Is it possible to include dual attributes (with old and new ID)?
@wrygiel What do you think about it?

**Comment by andrixnet** [comment]
DO you mean the same ID fo rhte same OC site written in 2 A codes?

**Comment by deg-pl** [comment]
No. One A-code assigned to two ID (new and old). But it was a question. I don't know that it is possible. I don't know OKAPI well.

**Comment by wrygiel** [comment]
> I'm afraid malfunctioning OKAPI during changes.

I will be leaving for holidays very soon, and I won't be able to help you much! However, you shouldn't be too afraid - only a couple of apps make use of attributes, so even if you break something, the impact will be minimal.

> No. One A-code assigned to two ID (new and old). But it was a question. I don't know that it is possible.

Yes, it's okay to assign many internal IDs to a single A-code.

**Comment by andrixnet** [comment]
The above mapping changes are set on OCRO and defined in OKAPI here https://github.com/opencaching/okapi/pull/444

**Comment by andrixnet** [comment]
Please notify when OCNL and OCPL make their updates.

Afterwards, the other attribute IDs which are / may be in conflict will be dealt with.

**Comment by andrixnet** [comment]
Related: https://github.com/opencaching/okapi/issues/445

**Comment by following5** [comment]
This issue must be solved, before translations can be eliminated from the database, which will allow to enable any language at any OC site.

PL, RO and NL meanwhile have compatible IDs. UK has [nine conflicting IDs](https://github.com/opencaching/opencaching-pl/blob/9e4c80d3b98aedbb0bb804d18f11749b137eb28f/lib/format.gpx.inc.php#L364). When the new DB update system is active, we can renumber those IDs. For example:

Access restrictions:
* 40 => 92: Not recommended at night
* 41 => 93: Not available during high tide
* 43 => 94: Nature reserve / Breeding season
* 44 => 95: Available during winter

Tools:
* 49 => 87: Climbing gear required
* 50 => 88: Cave equipment required
* 51 => 89: Scuba gear required

Cache type:
* 54 => 63: Investigation required
* 55 => 64: Mathematical problem

US may be done later in the same manner.

**Comment by harrieklomp** [comment]
Perhaps i am wrong but i like do a suggestion.
We use mostly GC attributes. We can change attribute ID in DB or later in code. For changing existing caches a SQLalter can be used to update them.
Why not use the same ID numbers like GC uses  and use OC attributes above ID -> 100?
For example: OC-ONLY attribute have ID 6 and could be 106 because GSAK knows it as ID 106.

**Comment by following5** [comment]
That's not possible, because GC attributes are three-state. E.g. "24/7" and "not 24/7" are both GC ID 13 (with additional flag "inc = 1" or "inc = 0"). For OC these are two attributes with different IDs.

**Comment by harrieklomp** [comment]
The GC ID is always as a Yes. "13 = Available at all times". For "Not 24/7" we can use ID 113 or even 913?

**Comment by following5** [comment]
Okay, we could do that. But why? It would be a huge renumbering, and I don't see a benefit. The only place where groundspeak IDs are needed are the three PHP scripts (including Okapi) where GPX files are created, and they already have translation tables for that.

When attribute refactoring is done, you will never again see any attribute IDs, except if you become a developer and want to change the GPX code. :-)

Those OC attribute IDs are just arbitrary numbers without any meaning.

**Comment by harrieklomp** [comment]
The OC has some kind of system that is used on the search page. Is it possible to use this when making a cache?
![selectie_292](https://user-images.githubusercontent.com/1060518/51431782-4993b100-1c2e-11e9-9fb6-937da969c927.png)
This is also i kind of "yes/no" thing.

**Comment by following5** [comment]
> Is it possible to use this when making a cache?

Yes, that's possible, but it is independend from what IDs the attributes have, and how they are stored internally. It's just a matter of user interface design.

**Comment by andrixnet** [comment]
Any changes in attributes anywhere results in a code/dependency/documentation/maintenance nightnamre, that is why I am working on unifying the numbering.

**Comment by following5** [comment]
The reason for the nightmare is not how the IDs are numbered, but the existence of a `cache_attrib` table which any sysadmin can edit. To get rid of the nightmare, attributes must be defined globally in code instead of locally in the database.

In preparation for that, I already created a unified ID set for all sites except US. The definitions are here:

https://github.com/opencaching/opencaching-pl/blob/master/lib/languages/en.php#L1923

Well, I run out of time to finish that. :( So maybe it is better that you continue to do it your way.

**Comment by andrixnet** [comment]
Work is well under way.
